### PR TITLE
Support frequency-domain loss

### PR DESCRIPTION
**Pull request by Emrys365**
This modification allows using Graph-PIT for frequency-domain losses, e.g.
```python
import torch
from graph_pit import graph_pit_loss

# Create three target utterances and two estimated signals
targets = [torch.rand(100, 257), torch.rand(200, 257), torch.rand(150, 257)]   # List of target utterance signals 
segment_boundaries = [(0, 100), (150, 350), (300, 450)]     # One start and end time for each utterance 
estimate = torch.rand(2, 500, 257)

# Compute loss with the unoptimized loss function, here mse for example
loss = graph_pit_loss(
    estimate, targets, segment_boundaries,
    torch.nn.functional.mse_loss
)
```

### CODE REVIEW

**Comment by thequilo** [comment]
Hi @Emrys365!

Your change makes sense, I didn't think of this case while implementing it. If you want to use the optimized Graph-PIT MSE loss, I just opened a PR that adds that: #2. The optimized variant is a lot faster for large numbers of utterances.

**Comment by Emrys365** [comment]
Thanks a lot! That's very cool :)
